sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -35,6 +35,9 @@ object UnsupportedOperationChecker {
       case p if p.isStreaming =>
         throwError("Queries with streaming sources must be executed with writeStream.start()")(p)
 
+      case p: Deduplication =>
+        throwError("Batch queries should not use Deduplication")(p)
```

Comment:
Why is deduplication exclusive to streaming?  Even if we don't want to implement a batch operator, I'd still allow it in the logical plan and just translate it to normal aggregation.